lib/tasks/search.rake

```diff
@@ -1,170 +1,36 @@
-      #  document won't create duplicates.
-      LOG.info "importing any documents that changed between #{start_time} and now"
-      cursor = Content.where(:_type.in => content_types, :updated_at.gte => start_time)
-      import_from_cursor(cursor, new_index, opts)
-    end
-  end
-
-  desc 'Copies contents of MongoDB into Elasticsearch if updated in the last N minutes.'
-  task :catchup, [:minutes, :batch_size, :sleep_time] => :environment do |t, args|
-    opts = batch_opts args
-    the_index = TaskHelpers::ElasticsearchHelper.get_index
-    alias_ = Tire::Alias.find the_index.name
-    # this check makes sure we are working with the index to which
-    # the desired model's alias presently points.
-    raise RuntimeError, "could not find live index" if alias_.nil?
+  desc 'Indexes content updated in the last N minutes.'
+  task :catchup, [:minutes] => :environment do |t, args|
     start_time = Time.now - (args[:minutes].to_i * 60)
-    cursor = Content.where(:_type.in => %w(Comment CommentThread), :updated_at.gte => start_time)
-    import_from_cursor(cursor, the_index, opts)
-  end
 
-  def batch_opts(args)
-    args = args.to_hash
-    {:batch_size => args[:batch_size].nil? ? 500 : args[:batch_size].to_i,
-     :sleep_time => args[:sleep_time].nil? ? 0 : args[:sleep_time].to_i}
+    [Comment, CommentThread].each do |model|
+      model.where(:updated_at.gte => start_time).import(index: Content::ES_INDEX_NAME)
+    end
   end
 
-  desc 'Removes any data from Elasticsearch that no longer exists in MongoDB.'
-  task :prune, [:batch_size, :sleep_time] => :environment do |t, args|
-    opts = batch_opts args
-    the_index = TaskHelpers::ElasticsearchHelper.get_index
-    puts "pruning #{the_index.name}"
-    alias_ = Tire::Alias.find the_index.name
-    raise RuntimeError, 'could not find live index' if alias_.nil?
-    scan_size = opts[:batch_size] / TaskHelpers::ElasticsearchHelper.get_index_shard_count(the_index.name)
-    cnt = 0
-    [CommentThread, Comment].each do |klass|
-      doc_type = klass.document_type
-      # this check makes sure we are working with the index to which
-      # the desired model's alias presently points.
-      search = Tire::Search::Scan.new the_index.name, {size: scan_size, type: doc_type}
-      search.each do |results|
-        es_ids = results.map(&:id)
-        mongo_ids = klass.where(:id.in => es_ids).map { |d| d.id.to_s }
-        to_delete = es_ids - mongo_ids
-        if to_delete.size > 0
-          cnt += to_delete.size
-          puts "deleting #{to_delete.size} orphaned #{doc_type} documents from elasticsearch"
-          the_index.bulk_delete (to_delete).map { |v| {"type" => doc_type, "id" => v} }
-        end
-        puts "#{the_index.name}/#{doc_type}: processed #{search.seen} of #{search.total}"
-        sleep opts[:sleep_time]
-      end
+  desc 'Reindex all data from the database'
+  task :reindex, [:index] => :environment do |t, args|
```

Comment:
@maxrothman: Do you want to review this change in rake tasks, even if just from a high level?  Since this is in use for Olive, I guess it is ok for you?  Do we need to manually test these?  Do you want to do this?
FYI: @edx/devops: Since Max is out.